File: C++/PowerOfThree.cc
// Given an integer, write a function to determine if it is a power of three.

class Solution {
     const vector<int> pow3{1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049, 177147, 531441, 1594323, 4782969, 14348907, 43046721, 129140163, 387420489, 1162261467};
public:
    bool isPowerOfThree(int n) {
        return binary_search(pow3.begin(), pow3.end(), n);
    }
};

File: Java/PowerOfThree.java
// Given an integer, write a function to determine if it is a power of three.

// Follow up:
// Could you do it without using any loop / recursion?

public class PowerOfThree {
    final int maxPow = (int) Math.pow(3, 19);
    public boolean isPowerOfThree(int n) {
        return n > 0 &&  maxPow % n == 0;
    }
}
